From 25f506c86f8a2e3c597bdfd5619ddda56df9028c Mon Sep 17 00:00:00 2001
From: Filip Drazic <filip.drazic@aggios.com>
Date: Fri, 10 Jun 2016 16:53:53 +0200
Subject: [PATCH 733/793] usb: dwc3: of-simple: Add missing
 platform_set_drvdata

Device driver data is accessed in driver's remove method.
This fixes the following error when trying to unbind device from
this driver:

Unable to handle kernel NULL pointer dereference at virtual address 00000010

Signed-off-by: Filip Drazic <filip.drazic@aggios.com>
Signed-off-by: Michal Simek <michal.simek@xilinx.com>
---
 drivers/usb/dwc3/dwc3-of-simple.c |    2 ++
 1 files changed, 2 insertions(+), 0 deletions(-)

diff --git a/drivers/usb/dwc3/dwc3-of-simple.c b/drivers/usb/dwc3/dwc3-of-simple.c
index e56d59b..834ab8c 100644
--- a/drivers/usb/dwc3/dwc3-of-simple.c
+++ b/drivers/usb/dwc3/dwc3-of-simple.c
@@ -98,6 +98,8 @@ static int dwc3_of_simple_probe(struct platform_device *pdev)
 		return ret;
 	}
 
+	platform_set_drvdata(pdev, simple);
+
 	pm_runtime_set_active(dev);
 	pm_runtime_enable(dev);
 	pm_runtime_get_sync(dev);
-- 
1.7.5.4

